### PR TITLE
Reservation

### PR DESCRIPTION
**Pull request by xtrahuman**
# Rails setup for reservations
> In this pull request
- I set up the controller for reservations create and index
- I set up routes for reservation
- I set up migration files and schema for reservations
- I setup model to validate and provide association for Reservations

### CODE REVIEW

File: app/controllers/api/reservations_controller.rb [hunk 1]
```diff
@@ -26,7 +25,7 @@ def create
     private
 
     def reserve_param
-      params.require(:reservation).permit(:start_date, :city)
+      params.require(:reservation).permit(:start_date, :city, :user_id)
```
Comment (Taher-web-dev) [inline app/controllers/api/reservations_controller.rb:28]: you should permit user_id and car_id in this function.

File: app/controllers/api/reservations_controller.rb [hunk 2]
```diff
@@ -8,9 +8,8 @@ def index
     end
 
     def create
-      reserve = Reservation.new(start_date: params[:start_date], city: params[:city])
+      reserve = Reservation.new(reserve_param)
       @car = Car.find(params[:car_id])
```
Comment (Taher-web-dev) [inline app/controllers/api/reservations_controller.rb:12]: delete this ligne of code ise useless when you handle them inside reserve_param function.

Comment (Taher-web-dev) [reply app/controllers/api/reservations_controller.rb:12]: delete this ligne of code ise useless when you handle them inside reserve_param function.

**Review by Taher-web-dev — CHANGES_REQUESTED** [review]
You should make your reserve _param handle all the data needed to create new reserve.

File: app/controllers/api/reservations_controller.rb (hunk 2 above)
Comment (xtrahuman) [reply app/controllers/api/reservations_controller.rb:12]: Thanks for the feed back but this how the URL for post request  would look like in frontend, which means we can get the id using find params
`http://localhost:3000/api/cars/{id}/reservations`

Comment (xtrahuman) [reply app/controllers/api/reservations_controller.rb]: this routes `resources :cars , only:[:index] do
resources reservations[:index, :create]`

was used to get this

`http://localhost:3000/api/cars/{id}/reservations`

I have used some api with routes this way on the frontend.

but i have made the changes as requested

**Pull request merged**